src/components/discover-sheet/DPICard.tsx

```diff
@@ -0,0 +1,164 @@
+import analytics from '@segment/analytics-react-native';
+import lang from 'i18n-js';
+import React, { useCallback } from 'react';
+import LinearGradient from 'react-native-linear-gradient';
+import { useSelector } from 'react-redux';
+import { useNavigation } from '../../navigation/Navigation';
+import { ButtonPressAnimation } from '../animations';
+import { CoinIcon } from '../coin-icon';
+import {
+  AccentColorProvider,
+  Bleed,
+  Box,
+  ColorModeProvider,
+  Column,
+  Columns,
+  Cover,
+  Heading,
+  Inset,
+  Stack,
+  Text,
+  useForegroundColor,
+} from '@rainbow-me/design-system';
+import { useAccountSettings } from '@rainbow-me/hooks';
+import { DPI_ADDRESS } from '@rainbow-me/references';
+import Routes from '@rainbow-me/routes';
+import { useTheme } from '@rainbow-me/theme';
+import { ethereumUtils } from '@rainbow-me/utils';
+
+export default function DPICard() {
+  const { nativeCurrency } = useAccountSettings();
+  const { navigate } = useNavigation();
+  const { genericAssets } = useSelector(({ data: { genericAssets } }) => ({
+    genericAssets,
+  }));
```

Comment:
Can we replace it with `useStore` call so we can basically use `store.getState().data.genericAssets` inside the handlePress. In that case, we won't be rerendering this component in the background every time we receive new genericAssets but still be using the latest version of genericAssets in the handler.